perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -2060,4 +2179,108 @@ private void closeExtSourcesAfterSynchronization(ExtSource membersSource, ExtSou
 			}
 		}
 	}
+	
+	@Override
+	public void processRelationMembers(PerunSession sess, Group resultGroup, List<Member> changedMembers, int sourceGroupId, boolean addition) throws GroupOperationsException {
+
+		List<Member> newMembers;
+
+		try {
+			if (addition) {
+				newMembers = addIndirectMembers(sess, resultGroup, changedMembers, sourceGroupId);
+			} else {
+				newMembers = removeIndirectMembers(sess, resultGroup, changedMembers, sourceGroupId);
+			}
+
+			if (newMembers.isEmpty()) {
+				return;
+			}
+
+			List<Integer> relations = groupsManagerImpl.getGroupRelations(sess, resultGroup.getId());
+			for (Integer groupId : relations) {
+				processRelationMembers(sess, groupsManagerImpl.getGroupById(sess, groupId), newMembers, resultGroup.getId(), addition);
+			}
+		} catch (WrongReferenceAttributeValueException | WrongAttributeValueException | AlreadyMemberException |
+				InternalErrorException | NotGroupMemberException | GroupNotExistsException ex) {
+			throw new GroupOperationsException(ex);
+		}
+	}
+
+	@Override
+	public Group createGroupUnion(PerunSession sess, Group resultGroup, Group operandGroup, boolean parentFlag) throws GroupOperationsException, InternalErrorException {
+		
+		//check if any of the groups is members group
+		if(resultGroup.getName().equals(VosManager.MEMBERS_GROUP) || operandGroup.getName().equals(VosManager.MEMBERS_GROUP)) {
+			throw new InternalErrorException("Union cannot be created on members group.");
```

Comment:
Unless there is another check which prevent user to call this method with MEMBERS_GROUP in parameter, another exception should be used.